### ISSUE TITLE
Trained model

### ISSUE THREAD

**Issue by WeedsInML** [zerchen/AlignSDF#3]
Hello.
This is very nice work.
Could you please provide your trained model?

**Comment by zerchen** [comment]
Hi,

Yes, I will upload my trained model soon. You could first give a try to use my preprocessed data (https://drive.google.com/drive/folders/1GoaA6vB6TwAAHmaobVo5GjRoCq2wT21R) to do the training.

Best,
Zerui

**Comment by EAST-J** [comment]
> Hi,
> 
> Yes, I will upload my trained model soon. You could first give a try to use my preprocessed data (https://drive.google.com/drive/folders/1GoaA6vB6TwAAHmaobVo5GjRoCq2wT21R) to do the training.
> 
> Best, Zerui

Hi,
Have you uploaded the trained model?

**Comment by zerchen** [comment]
Hi,

Thanks for your reminder. I have put the trained model on ObMan on [here](https://drive.google.com/drive/folders/1ZwQD0oMJslkTamzKiFhZ5uLQRF40LnG_). Please give a note if it does not work or not compatible with this codebase. The checkpoints trained on dexycb will be released with a new codebase, which is more user-friendly.

Best,
Zerui

**Comment by RNGrunshen** [comment]
> Hi,
> 
> Thanks for your reminder. I have put the trained model on ObMan on [here](https://drive.google.com/drive/folders/1ZwQD0oMJslkTamzKiFhZ5uLQRF40LnG_). Please give a note if it does not work or not compatible with this codebase. The checkpoints trained on dexycb will be released with a new codebase, which is more user-friendly.
> 
> Best, Zerui

hello how can I use the  trained model to evaluate

**Comment by happyiminjay1** [comment]
Hello, I found a mismatch while loading your trained model.

Can you provide the 'specs.json' ?

**Comment by happyiminjay1** [comment]
> Hi,
> 
> Thanks for your reminder. I have put the trained model on ObMan on [here](https://drive.google.com/drive/folders/1ZwQD0oMJslkTamzKiFhZ5uLQRF40LnG_). Please give a note if it does not work or not compatible with this codebase. The checkpoints trained on dexycb will be released with a new codebase, which is more user-friendly.
> 
> Best, Zerui

Hello,

I think the specs.json file which is uploaded on the experiments/obman/*.json file is not compatible to the trained model.

Can you upload the right specs.json file?

Thanks.

**Comment by zerchen** [comment]
Hi,

Thanks for your interests. Actually, I have not touched this codebase for a while. Some configurations might have changed during iterations of different versions.
I suggest using the latest [codebase](https://github.com/zerchen/gSDF), which provides a cleaned implementation of AlignSDF. To test the model, you could download the [directory](https://drive.google.com/drive/folders/16zDbN6W3DdtosogD6P2InU7arwdWRzl6) and place it under `$gSDF_ROOT/outputs`. Then, please follow the instructions written [here](https://github.com/zerchen/gSDF) to test and evaluate the model. Please note that the current checkpoint is only trained with 750 epochs, instead of 1600 epochs used in the original paper. More checkpoints will come later.

Best,
Zerui

**Comment by happyiminjay1** [comment]
Thank you for the clear explanation! I will try on the latest codebase.

Someone who needs the specs.json file can use the following text.

{
    "Description": "3D hand reconstruction on the mini obman dataset.",
    "DataSource": "data",
    "ImageSource": "rgb",
    "TrainSplit": "experiments/splits/obman_80k.json",
    "Dataset": "obman",
    "ModelType": "1encoder2decoder",
    "LMDB": true,
    "ImageSize": [256, 256],
    "SdfScaleFactor": 7.018621123357809,
    "LatentSize": 256,
    "PointFeatSize": 6,
    "EncodeStyle": "nerf",
    "ScaleAug": false,
    "PoseFeatSize": 15,
    "SnapshotFrequency": 500,
    "LogFrequency": 5,
    "LogFrequencyStep": 10,
    "NumEpochs": 1600,
    "Backbone": "resnet18",
    "Resume": "latest.pth",
    "Freeze": "none",
    "PixelAlign": false,
    "AdditionalSnapshots": [
      100,
      500
    ],
    "AdditionalLossStart": 1201,
    "SamplesPerScene": 2000,
    "ScenesPerBatch": 64,
    "DataLoaderThreads": 5,
    "ClampingDistance": 0.05,
    "HandBranch": true,
    "ObjectBranch": true,
    "ObjectPoseBranch": true,
    "ManoBranch": true,
    "DepthBranch": false,
    "Render": false,
    "ClassifierBranch": false,
    "ClassifierWeight": 0.005,
    "PenetrationLoss": false,
    "ContactLoss": false,
    "IndependentObjScale": false,
    "IgnorePointFromOtherMesh": false,
    "HandSdfWeight":0.5,
    "ObjSdfWeight":0.5,
    "JointWeight":0.5,
    "VertWeight":0,
    "ShapeRegWeight":0.0000005,
    "PoseRegWeight":0.00005,
    "SegWeight":0.004,
    "ObjCenterWeight":1,
    "ObjCornerWeight":0,
    "PenetrationLossWeight":15.0,
    "ContactLossWeight":0.005,
    "DisableAug":false,
    "BackgroundAug":false,
    "NetworkSpecs": {
      "dims": [
        512,
        512,
        512,
        512
      ],
      "dropout": [
        0,
        1,
        2,
        3
      ],
      "dropout_prob": 0.2,
      "norm_layers": [
        0,
        1,
        2,
        3
      ],
      "latent_in": [
        2
      ],
      "num_class": 6,
      "xyz_in_all": false,
      "use_tanh": false,
      "latent_dropout": false,
      "weight_norm": true
    },
    "LearningRateSchedule": [
      {
        "Type": "Step",
        "Initial": 0.0001,
        "Interval": 600,
        "Factor": 0.5
      }
    ]
  }
